### PR TITLE
feat(404): init for project environments services and clusters

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-806

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [ ] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview cc1de7de-94e5-40f8-8e45-bc8986d9dfec | [storybook](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/cc1de7de-94e5-40f8-8e45-bc8986d9dfec/services/general) |
 /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Programs must be written for people to read, and only incidentally for machines to execute

File: libs/domains/environments/feature/src/lib/hooks/use-environments/use-environments.ts [hunk 1]
```diff
@@ -82,6 +87,7 @@ export function useEnvironments({ projectId }: UseEnvironmentsProps) {
 
   return {
     data,
+    error: error as Error,
```
Comment (RemiBonnet) [inline libs/domains/environments/feature/src/lib/hooks/use-environments/use-environments.ts:90]: Necessary to get the correct type on the components side

File: libs/domains/projects/feature/src/lib/hooks/use-projects/use-projects.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ export function useProjects({ organizationId, enabled = true }: UseProjectsProps
       return data.sort((a, b) => a.name.localeCompare(b.name))
     },
     enabled,
+    retry: 3,
```
Comment (RemiBonnet) [inline libs/domains/projects/feature/src/lib/hooks/use-projects/use-projects.ts:19]: After 3 retry, we need to find an error to display 404

File: apps/console/src/app/router/main.router.tsx [hunk 3]
```diff
@@ -5,6 +5,7 @@ import { PageClusterCreateFeature, PageClusters } from '@qovery/pages/clusters'
 import { PageDatabase } from '@qovery/pages/database'
 import { PageEnvironments } from '@qovery/pages/environments'
 import { PageEvents } from '@qovery/pages/events'
+import { NotFoundPage } from '@qovery/pages/layout'
```
Comment (ctjhoa) [inline apps/console/src/app/router/main.router.tsx:8]: To remove

File: libs/pages/application/src/lib/page-application.tsx [hunk 4]
```diff
@@ -7,45 +7,52 @@ import {
...
+  const { data: serviceType, isSuccess: isSuccessServiceType } = useServiceType({
+    environmentId,
+    serviceId: applicationId,
+  })
   const { data: service } = useService({ environmentId, serviceId: applicationId })
 
-  useDocumentTitle(`${service?.name || 'Application'} - Qovery`)
+  useDocumentTitle(`${service?.name || 'Service'} - Qovery`)
+
+  if (environmentError || (!serviceType && isSuccessServiceType)) {
```
Comment (ctjhoa) [inline libs/pages/application/src/lib/page-application.tsx:31]:
```suggestion
 // serviceType can be `undefined` if the `find` method in `select` return nothing. However the query itself is still success.
 // Don't seems to have a better way for this case in react-query for now
 // https://github.com/TanStack/query/issues/1540
 // https://github.com/TanStack/query/issues/5878
  if (environmentError || (!serviceType && isSuccessServiceType)) {
```

File: libs/pages/application/src/lib/ui/container/container.tsx [hunk 5]
```diff
@@ -1,24 +1,41 @@
...
 
-export interface ContainerProps {
+export interface ContainerProps extends PropsWithChildren {
   service?: Exclude<AnyService, Database>
   environment?: Environment
 }
 
-export function Container({ service, environment, children }: PropsWithChildren<ContainerProps>) {
-  const { organizationId = '' } = useParams()
+export const ApplicationContext = createContext<{
```
Comment (ctjhoa) [inline libs/pages/application/src/lib/ui/container/container.tsx:25]: To remove

File: libs/pages/cluster/src/lib/page-cluster.tsx [hunk 6]
```diff
@@ -1,36 +1,41 @@
...
-          <Route key={route.path} path={route.path} element={route.component} />
-        ))}
-        <Route
-          path="*"
-          element={<Navigate replace to={CLUSTER_URL(organizationId, clusterId) + CLUSTER_SETTINGS_URL} />}
-        />
-      </Routes>
-    </Container>
-  )
+  if (!cluster && isFetched) {
```
Comment (ctjhoa) [inline libs/pages/cluster/src/lib/page-cluster.tsx:13]: prefer `isSuccess` + comment

File: libs/pages/cluster/src/lib/page-cluster.tsx [hunk 7]
```diff
@@ -1,36 +1,41 @@
...
+        <Routes>
+          {ROUTER_CLUSTER.map((route) => (
+            <Route key={route.path} path={route.path} element={route.component} />
+          ))}
+          <Route path="*" element={<NotFoundPage />} />
+        </Routes>
+      </Container>
+    )
+  } else {
+    return null
```
Comment (ctjhoa) [inline libs/pages/cluster/src/lib/page-cluster.tsx:37]: to remove

**Review by ctjhoa — CHANGES_REQUESTED** [review]

**Review by ctjhoa — APPROVED** [review]

**Pull request merged**